File: backend/content-service/src/main/java/com/educp/content/entity/Column.java
package com.educp.content.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.educp.common.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@TableName("columns")
@EqualsAndHashCode(callSuper = true)
public class Column extends BaseEntity {

    private String name;

    private String code;

    private String icon;

    private String routePath;

    @TableField("is_system")
    private Boolean isSystem;

    @TableField("is_visible")
    private Boolean isVisible;

    @TableField("sub_menu_enabled")
    private Boolean subMenuEnabled;

    private Integer sortOrder;

    private Long createdBy;
}

File: frontend/admin/src/types/column.ts
import type { SubColumn } from './subColumn'

export interface Column {
  id: number
  name: string
  code: string
  icon?: string
  routePath?: string
  isSystem: boolean
  isVisible: boolean
  subMenuEnabled: boolean
  sortOrder: number
  subColumns?: SubColumn[]
  createdAt: string
  updatedAt: string
}

export interface CreateColumnRequest {
  name: string
  code: string
  icon?: string
  routePath?: string
  sortOrder?: number
}

export interface UpdateColumnRequest {
  name?: string
  code?: string
  icon?: string
  routePath?: string
  isVisible?: boolean
  subMenuEnabled?: boolean
  sortOrder?: number
}

export interface SortColumnRequest {
  items: Array<{ id: number; sortOrder: number }>
}
